Ten Elephants/Ten Elephants/Cache/CachedImageFetcher.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  CachedImageFetcher.swift
+//  Ten Elephants
+//
+//  Created by Kirill Denisov on 13.12.2021.
+//
+
+import UIKit
+import Foundation
+
+class CachedImageFetcher {
+    
+    private enum Constants {
+        static let cacheCountLimit: Int = 25
+    }
+    
+    init() {
+        self.cachedImages.countLimit = Constants.cacheCountLimit
+    }
+    
+    init(countLimit: Int) {
+        self.cachedImages.countLimit = countLimit
+    }
+    
+    let cachedImages = NSCache<NSURL, UIImage>()
+    var loadingResponses = [NSURL: [(UIImage?) -> Swift.Void]]()
+
+    func image(url: NSURL) -> UIImage? {
+        cachedImages.object(forKey: url)
+    }
+
+    func fetch(url: NSURL, completion: @escaping (UIImage?) -> Swift.Void) {
+        if let cachedImage = image(url: url) {
+            DispatchQueue.main.async {
+                completion(cachedImage)
+            }
+            return
+        }
+        if loadingResponses[url] != nil {
+            loadingResponses[url]?.append(completion)
+            return
+        } else {
+            loadingResponses[url] = [completion]
+        }
+        let task = URLSession.shared.dataTask(with: url as URL) { (data, _, error) in
+            guard let responseData = data, let image = UIImage(data: responseData),
+                  let blocks = self.loadingResponses[url], error == nil else {
+                DispatchQueue.main.async {
+                    completion(nil)
+                }
+                return
+            }
+            self.cachedImages.setObject(image, forKey: url, cost: responseData.count)
```

Comment:
Мы не выставляешь ограничение по cost, есть смысл указывать эту стоимость при добавлении?
```suggestion
            self.cachedImages.setObject(image, forKey: url)
```